test/lit/passes/dae-gc-refine-return.wast

```diff
@@ -654,7 +658,9 @@
  ;; CHECK-NEXT:  (drop
  ;; CHECK-NEXT:   (call $tail-caller-call_ref-no)
  ;; CHECK-NEXT:  )
- ;; CHECK-NEXT:  (call $tail-caller-call_ref-unreachable)
+ ;; CHECK-NEXT:  (drop
+ ;; CHECK-NEXT:   (call $tail-caller-call_ref-unreachable)
+ ;; CHECK-NEXT:  )
```

Comment:
Why were drops added here?

Reply:
Previously the text parser used `ValidatingBuilder::validateAndMakeCallRef` for return calls and that would produce an unreachable block expression rather than an actual call inside `tail-caller-call_ref-unreachable`. Now the parser actually produces a tail call `CallRef` expression there that prevents DAE from removing the return value, so that return value needs to be dropped here.